auth/providers/azure/pop_tokenverifier_test.go

```diff
@@ -239,63 +241,68 @@ func GeneratePoPToken(ts int64, hostName, kid string) (string, error) {
 func TestPopTokenVerifier_Verify(t *testing.T) {
 	verifier := NewPoPVerifier("testHostname", 15*time.Minute)
 
-	validToken, _ := GeneratePoPToken(time.Now().Unix(), "testHostname", "")
+	validToken, _ := GeneratePoPToken(time.Now().Unix(), "testHostname", "", "randomnonce")
 	_, err := verifier.ValidatePopToken(validToken)
 	assert.NoError(t, err)
 
-	invalidToken, _ := GeneratePoPToken(time.Now().Unix(), "", badTokenKey)
+	// Test for replay attack
+	validToken, _ = GeneratePoPToken(time.Now().Unix(), "testHostname", "", "randomnonce")
```

Comment:
For a replay attack, don't regenerate the token. There can be a separate test which uses a different timestamp but same nonce.